pkg/controllers/node/health/events.go

```diff
@@ -0,0 +1,36 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package health
+
+import (
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+
+	"sigs.k8s.io/karpenter/pkg/events"
+)
+
+func UnhealthyCluster(node *corev1.Node, reason string) events.Event {
+	return events.Event{
+		InvolvedObject: node,
+		Type:           corev1.EventTypeWarning,
+		Reason:         "UnhealthyNodePool",
```

Comment:
There's no NodePool attached to this node, so I'm not sure this event makes sense.

Also, should we fire events for Nodes/NodeClaims that are owned by a NodePool? I think, yes. Ideally, we apply the same eventing logic to all nodes across the system